sklearn/linear_model/tests/test_least_angle.py

```diff
@@ -439,4 +439,70 @@ def test_lars_path_readonly_data():
     splitted_data = train_test_split(X, y, random_state=42)
     with TempMemmap(splitted_data) as (X_train, X_test, y_train, y_test):
         # The following should not fail despite copy=False
-        _lars_path_residues(X_train, y_train, X_test, y_test, copy=False)
\ No newline at end of file
+        _lars_path_residues(X_train, y_train, X_test, y_test, copy=False)
+
+
+def test_lars_path_positive_constraint():
+    # this is the main test for the positive parameter on the lars_path method
+    # the estimator classes just make use of this function
+
+    # we do the test on the diabetes dataset
+
+    # ensure that we get negative coefficients when positive=False
+    # for method 'lar' (default)
+    alpha, active, coefs = linear_model.lars_path(
+            diabetes['data'], diabetes['target'],
+            return_path=True,
+            method='lar',
+            positive=False)
+    assert_true(coefs.min() < 0)
+    # for method 'lasso'
+    alpha, active, coefs = linear_model.lars_path(
+            diabetes['data'], diabetes['target'],
+            return_path=True,
+            method='lasso',
+            positive=False)
+    assert_true(coefs.min() < 0)
+
+    # now let's restrict the solution to be positive
+    # for method 'lar' (default)
+    alpha, active, coefs = linear_model.lars_path(
+            diabetes['data'], diabetes['target'],
+            return_path=True,
+            method='lar',
+            positive=True)
+    assert_true(coefs.min() >= 0)
+    # for method 'lasso'
+    alpha, active, coefs = linear_model.lars_path(
+            diabetes['data'], diabetes['target'],
+            return_path=True,
+            method='lasso',
+            positive=True)
+    assert_true(coefs.min() >= 0)
+
+
+# now we gonna test the positive option for all estimator classes
+
+default_parameter = { 'fit_intercept' : False, }
+
+estimator_parameter_map = {
+        'Lars' : { 'n_nonzero_coefs' : 5, },
+        'LassoLars' : { 'alpha' : 0.1, },
+        'LarsCV' : {  },
+        'LassoLarsCV' : { },
+        'LassoLarsIC' : { },
+        }
+
+def test_estimatorclasses_positive_constraint():
+    # testing the transmissibility for the positive option of all estimator
+    # classes in this same function here 
+
+    for estname in estimator_parameter_map:
+        params = default_parameter.copy()
+        params.update(estimator_parameter_map[estname])
+        estimator = getattr(linear_model, estname)(positive=False, **params)
+        estimator.fit(diabetes['data'], diabetes['target'])
+        assert_true(estimator.coef_.min() < 0)
+        estimator = getattr(linear_model, estname)(positive=True, **params)
+        estimator.fit(diabetes['data'], diabetes['target'])
+        assert_true(min(estimator.coef_) >= 0)
```

Comment:
how long does it take to run?

can you add a test that compares the resets of Lasso and LassoLars with positive == True? there is already a test. Just update it to also test with positive=True.